### ISSUE TITLE
[MHV SM and Medications / Accessibility] Filter and sort functionality provided in different ways

### ISSUE THREAD

**Issue by sarahhortonadhoc** [department-of-veterans-affairs/va.gov-team#125156]
### 📝 User story

As a user, I want features to operate in a consistent way, so that I can learn to use the functionality and apply what I know across the platform. As an AT user, I want common features to be identified programmatically and operate in a consistent way.

### 📝 Problem statement

The Sort and Filter features are design and implemented in different ways across products.

### 🌌 Supporting screenshots or videos

<img width="659" height="733" alt="Screenshot of the filter and sort features on the Medications list page" src="https://github.com/user-attachments/assets/a7f1185d-15a0-42ee-8c1d-995a50d22f65" />

The Medications list page provides a Filter in an Accordion component. The filters are radio buttons. The Sort option is a separate feature, provided with a Select menu that automatically refreshes the results and moves focus to the results details.

<img width="740" height="741" alt="Screenshot of the filter and sort features on the Secure Messaging list page" src="https://github.com/user-attachments/assets/45b960be-1b3a-4d0f-aef2-ccc21e685526" />

The Medications list page provides a Filter feature in a section above the messages and the filters are in a disclosure widget. The filters are provided in Select menus. Apply filters moves focus to the results under the Filter feature. The Sort option is separate, provided with a Select menu that applies the sort after selecting the Sort button,  and moves focus to the results details.

### 👣 Reproduction steps

1. Navigate to the [Medications list page](https://staging.va.gov/my-health/medications?page=1) (username: (mhvmark.test@id.me).
2. Use the Tab key to focus on the Filters accordion; select Enter to open the accordion panel.
3. Use the arrow key to select a radio button option.
4. Use the Tab key to move focus to the Apply Filters button; select Enter to apply filters.
5. Notice focus is set to the results details.
6. Use the Shift/Tab key to navigate back to the Sort menu; use the Arrow key to select an option and select Enter to apply.
7. Notice focus is set to the results details.

1. Navigate to the [Messages list page](https://staging.va.gov/my-health/secure-messages/inbox) (username: (mhvmark.test@id.me).
2. Use the Tab key to focus on the Show Filters button; select Enter to open the Filters panel.
3. Use the Tab key to focus on the Category input.
4. Use the arrow key to select a menu option; select Enter to apply the option.
5. Use the Tab key to move focus to the Apply Filters button; select Enter to apply filters.
6. Notice focus is set to a section with number of matches and category details.
7. Use the Tab key to navigate to the Sort menu; use the Arrow key to select an option.
8. Use the Tab key to navigate to the Sort button; and select Enter to apply sort.
9. Notice focus is set to the results details.

### 🔗 Point of contact

@sarahhortonadhoc

### ⚠️ Severity level

3, Moderate. Should be fixed in 1-3 sprints post-launch.

#### Accessibility Standards

- [Guideline 3.2 Predictable: Make web pages appear and operate in predictable ways.](https://www.w3.org/WAI/WCAG22/Understanding/predictable)
  - [SC 3.2.4: Consistent Identification (Level AA)](https://www.w3.org/WAI/WCAG22/Understanding/consistent-identification)

#### Experience Standards
- [Consistency: User has a predictable and unchanging experience across all VA.gov and VA Mobile App services, tools, and pages](https://design.va.gov/about/experience-standards/#category-consistency)
  - [User experiences predictable interactions with components and patterns](https://design.va.gov/about/experience-standards/#user-experiences-predictable-interactions-with-components-and-patterns)

### 𝌣 Proposed solution or next steps

Revisit the filter and sort design and implementation.

### 𝌣 Supporting resources

- [Focus management](https://design.va.gov/accessibility/focus-management) — from VADS
- [Select component accessibility considerations](https://designsystem.digital.gov/components/select/#accessibility-select) — from USWDS
- [Select component](https://design.va.gov/components/form/select) — from VADS
- [SC 3.2.2 On Input (Level A)](https://www.w3.org/WAI/WCAG22/Understanding/on-input) — from WCAG
- [Accordion component](https://design.va.gov/components/accordion) — from VADS

### ✅ Acceptance criteria

- [ ] The system provides filter and sort functionality using common components and patterns.
- [ ] The user recognizes the feature and can operate it effectively.
- [ ] The keyboard user can accurately predict which element will receive focus.
- [ ] The screen reader user does not miss relevant content and functionality.

### ✅ Possible tasks

- [ ] Review, refine, and acknowledge feedback.
- [ ] Update ticket information (title, assignee, labels, project(s), etc.)
- [ ] Fix and/or document decisions made.
- [ ] Product manager to move ticket in their respective product board to 'QA/A11y Review' after reviewing documented decisions / validating fix.
- [ ] Accessibility specialist will close the ticket after reviewing documented decisions / validating the fix.

**Comment by sarahhortonadhoc** [comment]
Work on this ticket is on hold while working through [MHV Portal / Accessibility] Explore accessible alternatives to filter/sort feature [#128669](https://github.com/department-of-veterans-affairs/va.gov-team/issues/128669).
